pixeltable/store.py

```diff
@@ -427,6 +428,35 @@ def delete_rows(
         status = conn.execute(stmt)
         return status.rowcount
 
+    def dump_rows(self, version: int, filter_view: StoreBase, filter_view_version: int) -> Iterator[dict[str, Any]]:
+        filter_predicate = sql.and_(
+            filter_view.v_min_col <= filter_view_version,
+            filter_view.v_max_col > filter_view_version,
+            *[c1 == c2 for c1, c2 in zip(self.rowid_columns(), filter_view.rowid_columns())],
+        )
+        stmt = (
+            sql.select('*')
```

Comment:
Don't you want to retrieve only the columns that are visible in 'version'? (Saves on IO, plus you mentioned that the merge would be in both the horizontal and vertical axis.)

Reply:
Maybe? Though it seems like one advantage to using StoreTable is that we don't have to worry about making sure we have the right list of columns selected (if we introduce new types of system columns, for example). And any columns that are not present in the version being published will necessarily have values that are all null, so the storage overhead is bounded. Let's consider this as a possible future optimization.